Anlab.Mvc/Controllers/ReviewerController.cs

```diff
@@ -88,7 +88,48 @@ public async Task<IActionResult> Details(int id)
             await GetHistories(id, model);
 
             return View(model);
-        }
+        }
+
+        public async Task<IActionResult> EmailList(DateTime? start, DateTime? end, string emailType, string orderAction)
+        {
+            var model = new EmailListModel();
+            if (start == null && end == null)
+            {
+                model.Start = DateTime.UtcNow.ToPacificTime().Date.AddDays(-30);
+                model.End = DateTime.UtcNow.ToPacificTime().Date;
+            }
+            else
+            {
+                model.Start = start?.Date;
+                model.End = end?.Date;
+            }
+            model.EmailType = emailType ?? "PI";
+            model.OrderAction = orderAction ?? "Finalized";
+
+            //Could potentially use the history table, but them it needs a foreign key to the order table
+            var query = _context.MailMessages.Include(a => a.Order).ThenInclude(a => a.Creator).Include(a => a.Order).AsQueryable();
+            if (model.Start != null)
+            {
+                query = query.Where(a => a.CreatedAt >= model.Start.Value.Date.FromPacificTime());
+            }
+            if (model.End != null)
+            {
+                query = query.Where(a => a.CreatedAt <= model.End.Value.Date.AddDays(1).FromPacificTime());
+            }
+            if(model.OrderAction == "Confirmed")
+            {
+                query = query.Where(a => a.Subject == "Work Order Confirmation" || a.Subject.StartsWith("Work Request Confirmation"));
+            }
+            else
+            {
+                query = query.Where(a => a.Subject.StartsWith("Work Request Finalized"));

```

Comment:
this could get pretty gnarly if we don't put any indexes on subject.   especially since email tables get large.  Let's keep an eye on it and make sure it doesn't take a long time running on prod.